### PR TITLE
[Baremetal QE] Adding an image for use by the baremetal qe installation steps

### PR DESCRIPTION
**Pull request by aleskandro**
This PR is to add an image for use by the baremetal qe steps (to be updated after merging this).
It also includes some common tool that could be helpful during future debugging.

/cc @jhou1 @jadhaj

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master.yaml [hunk 1]
```diff
@@ -29,6 +29,25 @@ images:
...
+          nmap-ncat socat mtr bash net-tools bind-utils iperf3 tcpdump procps conntrack-tools \
+          iputils ethtool strace ipmitool" && \
+      mkdir -p ${HOME} && \
+      yum install --setopt=install_weak_deps=False --nodocs -y $INSTALL_PKGS && \
+      yum clean all && \
+      rm -rf /var/cache/yum/*
+    RUN wget -O /usr/local/bin/yq \
+      https://github.com/mikefarah/yq/releases/download/v4.30.4/yq_linux_$(uname -m \
+        | sed 's/x86_64/amd64/;s/aarch64/arm64/') && \
+      ansible-galaxy collection install -vvvv community.docker community.general && \
```
[the remark this answers is not in the document]
Comment (aleskandro) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master.yaml:47]: Done

**Comment by kuiwang02** [comment]
@aleskandro @jhou1 I just am aware of this PR to add image here when I meet the golang PR can not be merged (https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_openshift-tests-private/6370/pull-ci-openshift-openshift-tests-private-master-images/1595969882581635072)
the image built fails.

Frankly this image has nothing with the code of openshift-tests-private repo, but it impact the PR merged.
So, I suggest to remove it here because it will cause PR merged unstable.
this image is used for cluster setup. you could try to check where to build image for cluster setup, and move it to there. for example, upi-installer image, openstack-installer and baremetal-installer etc. for baremetal-installer which maybe is what you need, you could get it from ci-operator/config/openshift/installer/openshift-installer-master.yaml

As summary, 
1) this image has nothing with code of this repo and has nothing with case tests, and bring unstable PR merged and enlarge PR merged duration as well. 
2) this image is related installation, so I suggest to move it to the correct place.

by the way if it happens more times, I will remove it temp.

**Comment by kuiwang02** [comment]
more one failure: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_openshift-tests-private/6370/pull-ci-openshift-openshift-tests-private-master-images/1595977332961579008

**Comment by kuiwang02** [comment]
openshift-tests-private is private repo and branch protected, so it can not promte the image into ocp namespace.
it is why I put golang image with excluded_images.
So actually test-private-bm is not promted, and then the step which use it should be failing.

and also you can not put it into __image.yaml file because once it fails, it will cause the regression does not get the latest test code.

So, I suggest you
1, move it to correct place as suggestion last comment
or
2, you could enhance dev-scripts image to what you want.

**Comment by kuiwang02** [comment]
@aleskandro @jhou1 ^^

**Comment by kuiwang02** [comment]
I guss your workflow baremetal-lab-install-ipi will fail after PR https://github.com/openshift/release/pull/34212
if you do not want to put it in correct place, you could put it into verification-test repo which is public and could promote it.
and then the usage is same to verification-test image.

**Comment by kuiwang02** [comment]
@aleskandro @jhou1 just provide my conclusion:

current code has two side effectives:
1, it causes PR merged unstable. and it causes the PR merged duration longer
2, it does not work for steps which use test-private-bm image because it is not promoted (actually it can not be promoted for this private repo. I already mention the reason in above comments). so your workflow should not work

So, in order to make it, we have the following solution:
1, the first solution: put it into correct place or enhance dev-scripts image
    I prefer to put it into correct place. could refer to baremetal-installer image in ci-operator/config/openshift/installer/openshift-installer-master.yaml

2, the second solution: put it into verification-tests repo.
     it still has the side effective 1). but there is no much PR merged for this repo now than test-private repo.
     it has no the side effective 2)
     I make the PR for example https://github.com/openshift/release/pull/34313 FYI.
     note: please make sure the qe-installer-bm image generted before starting workflow. it means there is PR merged for verification-test repo after the code changed.

3, the  third solution: current solution.
      as said, it should not work.

For me, I prefer to solution 1.  if you can not make solution 1, solution 2 is acceptable although I do not perfer to it.

by the way, if current code impact PR merged seriously, I will remove it temp.

**Comment by aleskandro** [comment]
The dockerfile is good, and the image takes ~5m, concurrently with the others that need more, to build => No big impacts on merge time, and no promotion is needed.

`Resolving github.com (github.com)... failed: Name or service not known.`

have you checked with dptp if there are issues with the infra? 

Regards

**Comment by kuiwang02** [comment]
it is instantaneous error. if we try it more times, it could pass. I do not check it with dptp (maybe please you check it with dptp if you like. For me, I do not think it is necessary). 
but it really impact PR merged, so I still do not think it is good solution. 
it really should not put here.

Or you change the way to install yq to avoid github.com as workaround. (although you get workaround, I am fear other tool installation failing impact PR merged as well in the future.)

by the way, did you try the workflow baremetal-lab-install-ipi and it succeeded?

**Comment by kuiwang02** [comment]
actually the simplest and correct way is to enhance https://github.com/openshift-metal3/dev-scripts/blob/master/images/Dockerfile.ci. no need to create more one image.

**Comment by aleskandro** [comment]
That's a completely different thing. The image is not meant to be used by dev-scripts.

If that's the case, we will move it whenever the workflows will be ready. For now, I'm ok with having the image there.

If you still face the flakiness on the buildfarm clusters, just comment `/retest`.
\

**Comment by kuiwang02** [comment]
@aleskandro 
1, I do not think it is different thing. the original is to use dev-scripts, and in order to enhance debugging, you make new image to replace it. so I think it is same thing.

2, the image has nothing with the code of this repo. should not be here. late some one follows this way, maybe more and more images which is not related to the code will go into here. it will bring PR merged more and more unstable.
    so, you should find correct place to put it if you still need more one image.

3, more again: your workflow baremetal-lab-install-ipi with this image works? Per my understanding, it should not work (should meet image  not found error) because the solution does not work for the step.

cc @jhou1

**Comment by aleskandro** [comment]
> 1, I do not think it is different thing. the original is to use dev-scripts, and in order to enhance debugging, you make new image to replace it. so I think it is same thing.

It's not, the workflows are still under development and has never been something related to dev-scripts.

> 2, the image has nothing with the code of this repo. should not be here. late some one follows this way, maybe more and more images which is not related to the code will go into here. it will bring PR merged more and more unstable.
so, you should find correct place to put it if you still need more one image.

We are not meant to do this as a permanent approach, but we need a temporary and comfortable place to store the image. If openshift-tests-private is not, we'll move to another org/repo.

> 3, more again: your workflow baremetal-lab-install-ipi with this image works? Per my understanding, it should not work (should meet image not found error) because the solution does not work for the step.

It should when we will get to configure it in the openshift-tests-private repo configs.

**Comment by kuiwang02** [comment]
for 1, it replace dev-scripts. should do same thing finally although it is not now
for 2, the image has nothing with code of the repo. Just talk with Jian zhang, we do not allow such image into repo.
for 3, it should not work although it is in same repo, because it is in different configs file. you will meet the image not found issue.

After talking with my leader, I will remove it. actually my example https://github.com/openshift/release/pull/34313 solution works for you , it has no image not found issue. note: please ensure there is PR in verification-test repo merged after this code merged in release repo, it will ensure there is such image in ci namespace of build farm registry.

**Pull request merged**